src/applications/app.helpers.js

```diff
@@ -74,3 +74,111 @@ export function isParcel(appOrParcel) {
 export function objectType(appOrParcel) {
   return isParcel(appOrParcel) ? "parcel" : "application";
 }
+
+export function validateArgumentsTypeAPI(
+  appName,
+  applicationOrLoadingFn,
+  activityFn,
+  customProps
+) {
+  if (typeof appName !== "string" || appName.length === 0)
+    throw Error(
+      formatErrorMessage(
+        20,
+        __DEV__ &&
+          `The first argument to registerApplication must be a non-empty string 'appName'`
+      )
+    );
+
+  if (!applicationOrLoadingFn)
+    throw Error(
+      formatErrorMessage(
+        23,
+        __DEV__ && "The application or loading function is required"
+      )
+    );
+
+  if (typeof activityFn !== "function")
+    throw Error(
+      formatErrorMessage(
+        24,
+        __DEV__ && `The activityFunction argument must be a function`
+      )
+    );
+
+  if (
+    !!customProps &&
+    (typeof customProps !== "object" || Array.isArray(customProps))
+  )
+    throw Error(
+      formatErrorMessage(
+        22,
+        __DEV__ && "optional customProps must be an object"
+      )
+    );
+}
+
+export function validateRegisterObjectAPI(configurationObject) {
+  const validKeys = ["name", "load", "isActive", "customProps"];
+
+  const invalidPassedKeys = Object.keys(configurationObject).reduce(
+    (invalidKeys, prop) =>
+      validKeys.includes(prop) ? invalidKeys : invalidKeys.concat(prop),
+    []
+  );
+
+  if (invalidPassedKeys.length !== 0)
+    throw Error(
+      formatErrorMessage(
+        30,
+        __DEV__ &&
+          `The configuration object valid keys is: ${validKeys.join(
+            ", "
+          )}. Invalid keys: ${invalidPassedKeys.join(", ")}.`
+      )
+    );
+
+  validateArgumentsTypeAPI(
+    configurationObject.name,
+    configurationObject.load,
+    configurationObject.isActive,
```

Comment:
Yeah, I was thinking if sharing validations for both object and argument would be the best approach to reduce code duplication, but it would only work if they really shared the same interface, which they won't in the near future...
Since object configuration will grow more complex overtime, I'll try to isolate both validations, specific for argument and object!